apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/Claimable.tsx

```diff
@@ -0,0 +1,646 @@
+
+          <div className="w-full space-y-2">
+            <FormLabel>Allowlist</FormLabel>
+            <div className="flex flex-col gap-3">
+              {allowListFields.fields.map((fieldItem, index) => (
+                <div className="flex items-start gap-3" key={fieldItem.id}>
+                  <FormField
+                    control={form.control}
+                    name={`allowList.${index}.address`}
+                    render={({ field }) => (
+                      <FormItem className="grow">
+                        <FormControl>
+                          <Input
+                            placeholder="0x..."
+                            {...field}
+                            disabled={!props.isOwnerAccount}
+                          />
+                        </FormControl>
+                        <FormMessage />
+                      </FormItem>
+                    )}
+                  />
+                  <ToolTipLabel label="Remove address">
+                    <Button
+                      variant="outline"
+                      className="!text-destructive-text bg-background"
+                      onClick={() => {
+                        allowListFields.remove(index);
+                      }}
+                      disabled={!props.isOwnerAccount}
+                    >
+                      <Trash2Icon className="size-4" />
+                    </Button>
+                  </ToolTipLabel>
+                </div>
+              ))}
+
+              {allowListFields.fields.length === 0 && (
+                <Alert variant="warning">
+                  <CircleAlertIcon className="size-5 max-sm:hidden" />
+                  <AlertTitle className="max-sm:!pl-0">
+                    No allowlist configured
+                  </AlertTitle>
+                </Alert>
+              )}
+            </div>
+
+            <div className="h-1" />
+
+            <div className="flex gap-3">
+              <Button
+                variant="outline"
+                size="sm"
+                onClick={() => {
+                  // add admin by default if adding the first input
+                  allowListFields.append({
+                    address: "",
+                  });
+                }}
+                className="gap-2"
+                disabled={!props.isOwnerAccount}
+              >
+                <PlusIcon className="size-3" />
+                Add Address
+              </Button>
+            </div>
+          </div>
+
+          <div className="flex justify-end">
+            <Button
+              size="sm"
+              className="min-w-24 gap-2"
+              disabled={updateMutation.isPending || !props.isOwnerAccount}
+              type="submit"
+            >
+              {updateMutation.isPending && <Spinner className="size-4" />}
+              Update
+            </Button>
+          </div>
+        </div>
+      </form>{" "}
+    </Form>
+  );
+}
+
+const mintFormSchema = z.object({
```

Comment:
The `tokenId` check will always fail because form values are always string. 

Need to use `z.coerce.number().optional()` instead of `z.number().optional()` here

Reply:
actually it should be `z.coerce.number().int().gte(0).optional()` to enforce integer and 0>= value

Reply:
also same for quantity - `z.coerce.number().int().gte(0)`

Reply:
ok, looked more into how this value is used, and this is actually converted to bigInt when submitting the transaction. So this should just be saved as string